src/pysdmx/api/qb/gds.py

```diff
@@ -0,0 +1,158 @@
+"""Build GDS-REST structure queries."""
+
+from enum import Enum
+from typing import Literal, Optional, Sequence, Union
+
+import msgspec
+
+from pysdmx.api.qb.util import (
+    REST_ALL,
+    REST_LATEST,
+)
+from pysdmx.errors import Invalid
+
+
+class GdsType(Enum):
+    """The type of GDS metadata to be returned."""
+
+    GDS_AGENCY = "agency"
+    GDS_CATALOG = "catalog"
+    GDS_SDMX_API = "sdmxapi"
+    GDS_SERVICE = "service"
+    GDS_URN_RESOLVER = "urn_resolver"
+    ALL = REST_ALL
+    LATEST = REST_LATEST
+
+
+_RESOURCES = {
+    GdsType.GDS_AGENCY,
+    GdsType.GDS_CATALOG,
+    GdsType.GDS_SDMX_API,
+    GdsType.GDS_SERVICE,
+    GdsType.GDS_URN_RESOLVER,
+}
+
+
+class GdsQuery(msgspec.Struct, frozen=True, omit_defaults=True):
+    """A query for base GDS metadata.
+
+    Attributes:
+        artefact_type: The type of GDS metadata to be returned.
+        agency: The agency (or agencies) maintaining the artefact(s)
+            to be returned.
+        resource_id: The resource ID(s) to query. Defaults to '*'.
+        version: The version(s) of the resource. Defaults to '*'.
+        resource_type: The type of resource (e.g., 'data', 'metadata').
```

Comment:
`resource_type: Filters the endpoints that support the requested resource type (eg, 'data', 'metadata')`